src/python/pants/core/util_rules/strip_source_roots.py

```diff
@@ -91,7 +91,7 @@ def determine_source_root(path: str) -> str:
 
     if request.representative_path is not None:
         source_root = determine_source_root(request.representative_path)
-        if source_root == "":
+        if source_root == ".":
```

Comment:
```suggestion
        if source_root == "/":
```

(I think)

Reply:
No - *patterns* can use `/` to anchor the buildroot, but a SourceRoot always contains a relpath (as is documented there).